lib/model/model.go

```diff
@@ -2492,17 +2496,15 @@ func (m *Model) CommitConfiguration(from, to config.Configuration) bool {
 
 	// Some options don't require restart as those components handle it fine
 	// by themselves.
-	from.Options.URAccepted = to.Options.URAccepted
-	from.Options.URSeen = to.Options.URSeen
-	from.Options.URUniqueID = to.Options.URUniqueID
-	from.Options.ListenAddresses = to.Options.ListenAddresses
-	from.Options.RelaysEnabled = to.Options.RelaysEnabled
-	from.Options.UnackedNotificationIDs = to.Options.UnackedNotificationIDs
-	from.Options.MaxRecvKbps = to.Options.MaxRecvKbps
-	from.Options.MaxSendKbps = to.Options.MaxSendKbps
-	from.Options.LimitBandwidthInLan = to.Options.LimitBandwidthInLan
-	from.Options.StunKeepaliveS = to.Options.StunKeepaliveS
-	from.Options.StunServers = to.Options.StunServers
+
+	// Copy fields that do not have the field set to true
+	util.CopyMatchingTag(&from.Options, &to.Options, "restart", func(v string) bool {
+		if len(v) > 0 && v != "true" {
+			panic(fmt.Sprintf(`unexpected struct value: %s. expected untagged or "true"`, v))
```

Comment:
Why be so strict in these checks? A simple `return v != "true"` (and `return v == "true"` in the folder case above) should accomplish the same and not suddenly panic once someone decides to be extra explicit in the config struct for whatever reason

Reply:
To prevent confusion, as in one place you want falses in the other you want trues, due to the imbalanced number of things.

Reply:
Hmm OK. Just as long as this is exercised by one of the tests (I didn't check, all good if it already is) so it doesn't blow up later in production if someone does it wrong.

Reply:
Well, we do save config while in the model and start and stop folders, so I am pretty sure it's covered.